### PR TITLE
chore(project): add setup for test failure with unexpected console

### PR DESCRIPTION
**Pull request by joshblack**
This PR adds some config to our Jest environment to fail tests that emit `console.{error,warn}` while executing. In CI, this setup will also fail tests that emit `console.log`. This is useful for cleaning up our test output, in addition to pointing to inconsistencies in our tests where props may be incorrectly applied.

In order to get the tests to go green, I also needed to update tests that failed these new checks. These tests more often than not were missing props, or adding props to DOM nodes that were meant for specific child components.

#### Changelog

**New**

- Add setup in Jest that fails on `consle.{error,warn,log}` statements (`log` is only in CI so that developers can debug locally)

**Changed**

- Update tests that failed `console.*` checks

**Removed**

### CODE REVIEW

**Comment by joshblack** [comment]
Dependant on: https://github.com/carbon-design-system/carbon/pull/5183

File: packages/components/tests/styles-test.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const files = glob.sync('**/*.scss', {
 const render = promisify(sass.render);
 
 describe('styles', () => {
-  jest.setTimeout(8000);
+  jest.setTimeout(10000);
```
Comment (tw15egan) [inline packages/components/tests/styles-test.js]: FWIW I had to change this to `12000` to not get an error 🤷‍♂

Comment (joshblack) [reply packages/components/tests/styles-test.js]: Ooof, same. Let me bump up to 20000 just in case

Comment (joshblack) [inline packages/components/tests/styles-test.js]:
```suggestion
  jest.setTimeout(20000);
```

**Review by tw15egan — APPROVED** [review]
LGTM! 👍 ✅

**Review by asudoh — APPROVED** [review]
LGTM basically, just wondered if utilizing PR# 4045 in https://github.com/facebook/jest is desirable.

**Comment by joshblack** [comment]
@asudoh yeah! `spyOn` is perfect, the mock just needs to be restored after being used. I think `DatePicker` is a good example: https://github.com/carbon-design-system/carbon/pull/5184/files#diff-209ead5f5d1e9733be537c4a0684ef04R333

**Comment by asudoh** [comment]
Just in case it wasn't clear enough - I meant auto-restoring mocks (e.g. upon global after-each).

**Pull request merged**